src/shared/mount-util.c

```diff
@@ -1923,7 +1925,19 @@ static int path_get_mount_info_at(
                 if (mnt_fs_get_id(fs) != mnt_id)
                         continue;
 
-                _cleanup_free_ char *fstype = NULL, *options = NULL;
+                _cleanup_free_ char *source = NULL, *target = NULL, *fstype = NULL, *options = NULL;
+
+                if (ret_source) {
+                        source = strdup(strempty(mnt_fs_get_source(fs)));
+                        if (!source)
+                                return log_oom_debug();
+                }
+
+                if (ret_target) {
+                        target = strdup(strempty(mnt_fs_get_target(fs)));
```

Comment:
the strempty() here is weird. mnt_fs_get_target() must return a valid path, otherwise weird stuff is happening. maybe add an assert here, or fail, but definitely don't tape over it